File: csharp/Agentic.Commerce.Protocol/Models/RiskSignalAction.cs
// this file is @generated
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Agentic.Commerce.Protocol
{
    [JsonConverter(typeof(StringEnumConverter))]
    public enum RiskSignalAction
    {
        [EnumMember(Value = "blocked")]
        Blocked = 1,

        [EnumMember(Value = "manual_review")]
        ManualReview = 2,

        [EnumMember(Value = "authorized")]
        Authorized = 3,
    }
}

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/RiskSignalAction.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class RiskSignalAction : ToQueryParam {
    @SerialName("blocked") BLOCKED,
    @SerialName("manual_review") MANUAL_REVIEW,
    @SerialName("authorized") AUTHORIZED;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
